modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeSystem.java

```diff
@@ -82,6 +84,144 @@ public int getDefaultPrecision(SqlTypeName typeName) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public RelDataType deriveDecimalDivideType(RelDataTypeFactory typeFactory,
+            RelDataType type1, RelDataType type2) {
+        // the only case in enlarging such a type in Number#MIN_VALUE / -1 operation
+        if (isIntType(type1) && isIntType(type2)) {
+            boolean typesNullability = type1.isNullable() || type2.isNullable();
+
+            switch (type1.getSqlTypeName()) {
```

Comment:
instead of copy-pasting all these switch cases it's better to do following:

1. get least restrictive type among arguments (`typeFactory.leastRestrictive(List.of(type1, type2))`)
2. get next type for the result type from p1